trillian/migrillian/election/etcd/election.go

```diff
@@ -76,27 +56,18 @@ func (e *Election) Await(ctx context.Context) (context.Context, error) {
 		cancel()
 		return nil, ctx.Err()
 	case rsp, ok := <-ch:
-		if !ok {
-			cancel()
-			return nil, errMasterUnconfirmed
-		}
-		if string(rsp.Kvs[0].Value) != e.instanceID {
+		if !ok || string(rsp.Kvs[0].Value) != e.instanceID {
+			// Mastership has been overtaken in the meantime, or not capturead at all.
 			cancel()
-			return nil, errMasterOvertaken
+			return cctx, nil
```

Comment:
I'm a bit suspicious here: I see that the `cctx` is cancelled before it's returned, so in theory everything should be fine if the caller is well-behaved(?), but up in `migrillian/core/controller.go#L97` returning these params is going skip the `err != nil` flow and cause a log line to be printed saying that this instance is the master for xxx tree, which seems unlikely to be the case if we're returning here.

Reply:
Good observation. I added a `mctx.Err` check before running as master.
Are you otherwise okay with returning a closed context? For me it seemed like an elegant thing to do (compared to having a hard-coded error), and also not being the master doesn't feel like an error for something named `Observe` (its idea is not forcing mastership but only reporting the status).